### ISSUE TITLE
  Default sigma schedule has a duplicate terminal sigma, causing the last Euler step to be a no-op

### ISSUE THREAD

**Issue by chachachachachaxxxxx** [Tencent-Hunyuan/Hunyuan3D-2.1#209]
The default sigma schedule in `Hunyuan3DDiTFlowMatchingPipeline` appears to produce one redundant denoising step.
  
In the pipeline, default sigmas are created with:

  ```python
  sigmas = np.linspace(0, 1, num_inference_steps) if sigmas is None else sigmas
  ```

  Reference:

  https://github.com/Tencent-Hunyuan/Hunyuan3D-2.1/blob/main/hy3dshape/hy3dshape/pipelines.py#L736-L744

  `np.linspace(0, 1, num_inference_steps)` includes both endpoints, so the final sigma is already `1.0`.

  Then `FlowMatchEulerDiscreteScheduler.set_timesteps()` appends another terminal `1.0`:

  ```python
  self.sigmas = torch.cat([sigmas, torch.ones(1, device=sigmas.device)])
  ```

  Reference:

  https://github.com/Tencent-Hunyuan/Hunyuan3D-2.1/blob/main/hy3dshape/hy3dshape/schedulers.py#L214-L218

  During Euler stepping, the update uses adjacent sigma differences:

  ```python
  sigma = self.sigmas[self.step_index]
  sigma_next = self.sigmas[self.step_index + 1]
  prev_sample = sample + (sigma_next - sigma) * model_output
  ```

  Reference:

  https://github.com/Tencent-Hunyuan/Hunyuan3D-2.1/blob/main/hy3dshape/hy3dshape/schedulers.py#L304-L307

  As a result, the final default step is:

  ```text
  1.0 -> 1.0
  ```

  So:

  ```python
  sigma_next - sigma == 0
  prev_sample == sample
  ```

  With `num_inference_steps=50`, this performs 50 model evaluations but only 49 effective Euler updates.

  ## Expected Behavior

  The default schedule should avoid duplicate adjacent terminal sigmas, so `num_inference_steps` corresponds to the number of effective Euler updates.

  ## Possible Fix

  One option is to make the pipeline default schedule open-ended:

  ```python
  sigmas = np.linspace(0, 1, num_inference_steps, endpoint=False) if sigmas is None else sigmas
  ```

  The scheduler can still append the terminal `1.0`, giving `num_inference_steps` non-zero Euler intervals while preserving the “start from 0” behavior.

  Custom `sigmas` behavior can remain unchanged.
